packages/connections/src/components/connections.spec.tsx

```diff
@@ -179,27 +197,41 @@ describe('Connections Component', function () {
   describe('connecting to a connection that is not succeeding', function () {
     let savedConnectableId: string;
     let savedUnconnectableId: string;
+    let tmpDir: string;
+
+    before(async function () {
+      tmpDir = fs.mkdtempSync(path.join(os.tmpdir(), 'connections-tests'));
+      TestBackend.enable(tmpDir);
 
-    beforeEach(async function () {
       savedConnectableId = uuid();
       savedUnconnectableId = uuid();
       writeFakeConnection(tmpDir, {
         _id: savedConnectableId,
         port: 27018,
       });
-      writeFakeConnection(
-        tmpDir,
-        await convertConnectionInfoToModel({
-          id: savedUnconnectableId,
-          connectionOptions: {
-            // Hopefully nothing is running on this port.
-            // Times out in 5000ms.
-            connectionString:
-              'mongodb://localhost:28099/?connectTimeoutMS=5000&serverSelectionTimeoutMS=5000',
-          },
-        })
-      );
 
+      const connectionModel = await convertConnectionInfoToModel({
+        id: savedUnconnectableId,
+        connectionOptions: {
+          // Hopefully nothing is running on this port.
+          // Times out in 5000ms.
+          connectionString:
+            'mongodb://localhost:28099/?connectTimeoutMS=5000&serverSelectionTimeoutMS=5000',
+        },
+      });
+      writeFakeConnection(tmpDir, connectionModel);
```

Comment:
Is unfortunate that `ConnectionStorage` being implemented on top of `ampersand` and `storage-mixin` doesn't easily allow to just set a different folder or to disable the keyring.

To be honest I'm not a big fan of what we are doing here. With this approach we would need to keep `writeFakeConnection` up-to-date with the storage format, basically replicating code from `ConnectionStorage` and leaking here all sort of internals.

What about we mock `ConnectionStorage` entirely instead? We have the part that reads / writes from files tested already in the data service and we will have some e2e tests around loading / writing connections. This would also brute-solve the flake.

Another clean alternative is to change the interface of `ConnectionStorage` to add 2 options `{basePath, useKeychain}` and use directly a `ConnectionStorage` instance to create fixtures. We probably need to change a thing or 2 in the connection model in that case.

Instead of this:

```
const ExtendedConnection = Connection.extend(storageMixin, {
```

we would need to do:

```
const createConnectionModel = (basePath, backend) => {
   return Connection.extend(storageMixin, {
  idAttribute: '_id',
  namespace: 'Connections',
  storage: {
    backend: backend
    namespace: 'Connections',
    basepath: basePath,
  },
  // ...
}

module.export = createConnectionModel(basepath, 'splice-disk-ipc');
```

Then we can use `createConnectionModel` also in `ConnectionStorage` and switch to the `test` backend if useKeychain is false.

Reply:
Good thinking, went ahead and mocked the backend for these tests.